gdk4/src/toplevel.rs

```diff
@@ -1,18 +1,53 @@
 // Take a look at the license at the top of the repository in the LICENSE file.
 
-use crate::{Event, Toplevel};
+use crate::{Event, Toplevel, ToplevelSize};
+use glib::signal::connect_raw;
+use glib::signal::SignalHandlerId;
 use glib::translate::*;
 use glib::IsA;
+use std::boxed::Box as Box_;
+use std::mem::transmute;
 
 pub trait ToplevelExtManual {
     #[doc(alias = "gdk_toplevel_inhibit_system_shortcuts")]
     fn inhibit_system_shortcuts<P: AsRef<Event>>(&self, event: Option<&P>);
 
     #[doc(alias = "gdk_toplevel_show_window_menu")]
     fn show_window_menu<P: AsRef<Event>>(&self, event: &P) -> bool;
+    fn connect_compute_size<F: Fn(&Toplevel, &mut ToplevelSize) + 'static>(
+        &self,
+        f: F,
+    ) -> SignalHandlerId;
 }
 
 impl<O: IsA<Toplevel>> ToplevelExtManual for O {
+    fn connect_compute_size<F: Fn(&Toplevel, &mut ToplevelSize) + 'static>(
+        &self,
+        f: F,
+    ) -> SignalHandlerId {
+        unsafe extern "C" fn compute_size_trampoline<
+            F: Fn(&Toplevel, &mut ToplevelSize) + 'static,
+        >(
+            this: *mut ffi::GdkToplevel,
+            size: *mut ffi::GdkToplevelSize,
+            f: glib::ffi::gpointer,
+        ) {
+            let f: &F = &*(f as *const F);
+            f(&from_glib_borrow(this), &mut from_glib_none(size))
```

Comment:
This won't work like this. You're creating a new `ToplevelSize` here and that is then modified by the callback.

```suggestion
            f(&from_glib_borrow(this), &mut *(size as *mut TopLevelSize))
```

This should work and you don't need any of `From` translation traits then